preview/src/js/components/SelectTemplate.js

```diff
@@ -0,0 +1,34 @@
+const React = require('react');
+
+module.exports = React.createClass({
+    handleSelectChange: function (e) {
+        if (e.target.value) {
+            this.value = e.target.value;
+        } else {
+            delete this.value;
+        }
+
+        this.props.onSelectFormChange(this.value || '');
+    },
+
+    render: function () {
+        const listItems = this.props.templates.map(function (template) {
+            return (
+                <option value={template}>
+                    {template}
+                </option>
+            );
+        }.bind(this));
```

Comment:
good to know, I removed the 'bind' anyway as the 'this' context wasn't even used in the function passed to map.